core/src/main/java/org/apache/brooklyn/util/core/osgi/Osgis.java

```diff
@@ -181,11 +181,11 @@ public BundleFinder requiringFromUrl(String url) {
         public List<Bundle> findAll() {
             boolean urlMatched = false;
             List<Bundle> result = MutableList.of();
-            for (Bundle b: framework.getBundleContext().getBundles()) {
+            allBundles: for (Bundle b: framework.getBundleContext().getBundles()) {
                 if (symbolicName!=null && !symbolicName.equals(b.getSymbolicName())) continue;
                 if (version!=null && !Version.parseVersion(version).equals(b.getVersion())) continue;
                 for (Predicate<? super Bundle> predicate: predicates) {
-                    if (!predicate.apply(b)) continue;
+                    if (!predicate.apply(b)) continue allBundles;
```

Comment:
+1 i don't like labels here, and in any case i think the inner loop as svet suggests can be replaced by one line

    if (!Predicates.and(predicates).apply(b)) continue;